### PR TITLE
client: fix image copy

### LINKED ISSUE #3610: Filed remote image download
The template below is mostly useful for bug reports and support questions.
Feel free to remove anything which doesn't apply to you and add more information where it makes sense.

# Required information

 * Distribution: Ubuntu
 * Distribution version: 16.04
 * The output of "lxc info" or if that fails:
   * Kernel version: 4.10.0-28-generic
   * LXC version: 2.0.8
   * LXD version: 2.16
   * Storage backend in use: zfs

# Issue description

Can't copy an image to another LXD server. It worked some weeks ago, but it seem a recent update introduced the issue. 

# Steps to reproduce
Log into **srv01** and try to copy the image to **srv02**.
```
$ lxc image copy imageX srv02: --alias imageX
```
Result:

```
error: Failed remote image download:
 - https://88.198.22.111:8443: not found
 - https://[2a01:4f8:130:3395::2]:8443: Get https://[2a01:4f8:130:3395::2]:8443/1.0: Unable to connect to: [2a01:4f8:130:3395::2]:8443
 - https://10.0.3.1:8443: Get https://10.0.3.1:8443/1.0: x509: certificate is valid for srv02, not srv01
```

### PR DESCRIPTION
**Pull request by albertodonato**
Fixes #3610

### CODE REVIEW

File: lxc/image.go [hunk 1]
```diff
@@ -388,29 +388,13 @@ func (c *imageCmd) run(conf *config.Config, args []string) error {
...
-		} else {
-			// Don't fetch full image info if we don't need aliases (since it's
-			// an expensive operation)
-			imgInfo = &api.Image{}
-			imgInfo.Fingerprint = image
-			imgInfo.Public = true
+		image := c.dereferenceAlias(d, inName)
+		imgInfo, _, err := d.GetImage(image)
+		if err != nil {
+			return err
```
Comment (stgraber) [inline lxc/image.go]: And now simplestreams will take forever again...

I'll send another branch that takes care of all the cases.

Comment (brauner) [inline lxc/image.go]: @stgraber sorry, seemed like an obvious fix. Didn't think about simplestreams...

Comment (stgraber) [inline lxc/image.go]: Yeah, that piece of code is never an obvious fix. We've been going back and forth over the same 10 lines at least 2-3 times now over the past month :)

Comment (brauner) [inline lxc/image.go]: I'll just relay any further decisions to merge to you from now on. ;)

Comment (stgraber) [inline lxc/image.go]: I'll add a bunch of comments this time around so that whoever looks at the code next will understand what we're doing :)

Comment (brauner) [inline lxc/image.go]: Excellent :)

**Comment by monkiki** [comment]
Thanks for the fix. When is it going to be available in the Ubuntu LXD PPA?

**Comment by stgraber** [comment]
I'm probably going to do a bugfix upload to Ubuntu and the PPA late this week or early next week.

I built you a custom LXD client that you can download here:
 - https://dl.stgraber.org/lxc-3610 (sha256: ee111756c5c5d7e56ed3397ca297d3555384e84f2f0816912323d347a7c61dfc)

It includes this fix and all other potential cherry-picks for the next Ubuntu upload.

**Comment by monkiki** [comment]
Tested and works fine: ```Image copied successfully!```
Great and thanks! :+1:

**Pull request merged**